user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/download/SampleInformationXLSXProvider.java

```diff
@@ -0,0 +1,185 @@
+
+
+  private static void setAutoWidth(Sheet sheet) {
+    for (int col = 0; col <= SamplePreviewColumn.values().length; col++) {
+      sheet.autoSizeColumn(col);
+    }
+  }
+
+  private static void formatHeader(Row header, Map<String, Integer> variableNamesToColumn) {
+    for (SamplePreviewColumn value : SamplePreviewColumn.values()) {
+      var cell = header.createCell(value.column());
+      cell.setCellValue(value.header());
+    }
+    for (Map.Entry<String, Integer> entry : variableNamesToColumn.entrySet()) {
+      var cell = header.createCell(entry.getValue());
+      cell.setCellValue(entry.getKey());
+    }
+  }
+
+  public Set<String> uniqueExperimentalVariables(List<SamplePreview> samples) {
+    HashSet<String> variableNames = new HashSet<>();
+    for (SamplePreview sample : samples) {
+      sample.experimentalGroup().condition().getVariableLevels()
+          .forEach(vl -> variableNames.add(vl.variableName().value()));
+    }
+    return variableNames;
+  }
+
+  @Override
+  public byte[] getContent() {
+    if (samples.isEmpty()) {
+      return new byte[0];
+    }
+
+    ByteArrayOutputStream byteArrayOutputStream;
+
+    try (Workbook workbook = new XSSFWorkbook()) {
+      Sheet sheet = workbook.createSheet("Sample Information");
+
+      Row row = sheet.createRow(0);
+      Map<String, Integer> variableNamesToColumn = new HashMap<>();
+      Set<String> experimentalVariableNames = uniqueExperimentalVariables(samples);
+      Iterator<String> variableNamesIterator = experimentalVariableNames.iterator();
+      for (int assignedColumn = SamplePreviewColumn.values().length;
+          assignedColumn < SamplePreviewColumn.values().length + experimentalVariableNames.size(); assignedColumn++) {
+        variableNamesToColumn.put(variableNamesIterator.next(), assignedColumn);
+      }
+      formatHeader(row, variableNamesToColumn);
+
+      int rowNum = 1;
+
+      for (SamplePreview sample : samples) {
+        Row sampleRow = sheet.createRow(rowNum);
+        createSampleInfoEntry(sample, sampleRow, variableNamesToColumn);
+        rowNum++;
+      }
+
+      setAutoWidth(sheet);
+      byteArrayOutputStream = new ByteArrayOutputStream();
+      workbook.write(byteArrayOutputStream);
+
+
+    } catch (IOException e) {
+      log.error(e.getMessage(), e);
+      throw new ApplicationException(ErrorCode.GENERAL, null);
+    }
+
+    return byteArrayOutputStream.toByteArray();
+  }
+
+  private void createSampleInfoEntry(SamplePreview sample, Row sampleRow,
+      Map<String, Integer> variableNamesToColumn) {
+    var sampleIDcol = sampleRow.createCell(SamplePreviewColumn.SAMPLE_ID.column());
+    sampleIDcol.setCellValue(sample.sampleCode());
+
+    var labelCol = sampleRow.createCell(SamplePreviewColumn.LABEL.column());
+    labelCol.setCellValue(sample.sampleLabel());
+
+    var organismIdCol = sampleRow.createCell(SamplePreviewColumn.ORGANISM_ID.column());
+    organismIdCol.setCellValue(sample.organismId());
```

Comment:
The set cell value method sets the cell value to be a text even if it is a number.